### DISCUSSION TITLE
a small question about code, want to deeply understand

### DISCUSSION THREAD

**Discussion by yuantangjin** [prashjha/PeriDEM#19] [Q&A] [answered]
https://github.com/prashjha/PeriDEM/blob/22f97a5e79347a98cc3022253bbe3f4f387427ba/src/inp/input.cpp#L53
```
if (!d_modelDeck_p->d_isPeriDEM)
    setMeshDeck();
if (!d_modelDeck_p->d_isPeriDEM)
    setMaterialDeck();
```
why not：
```
if (!d_modelDeck_p->d_isPeriDEM)
{
    setMeshDeck();
    setMaterialDeck();
}
```
What are your considerations for using the first method of writing above? Is it because of parallel computing?

**Comment by prashjha** [comment, marked as answer]
Second one is more efficient. I don't recall why I used first method. There maybe more pieces like these and these are from the first iteration of code. Ideally I should relook at all files for optimization but that will take long time.

**Reply by yuantangjin** [reply]
Ok，I understand. I can check it again on your behalf, and I will communicate with you via email if there is any technical problem. However, I may still need your help in understanding some aspects. It won't take you too much time.
Thanks for your reply😀
